File: api/models/Doctor.php
<?php

    $ds = DIRECTORY_SEPARATOR;
    $base_dir = realpath(dirname(__FILE__). $ds . '..') . $ds;

    require_once("{$base_dir}includes{$ds}Database.php"); //including Database

    class Doctor{
        private $table = 'doctors';

        public $Doctor_ID;
        public $Doctor_Name;
        public $Doctor_Email;
        public $Doctor_Address;
        public $Doctor_Contact;
        public $Doctor_Country;
        public $Doctor_State;
        public $Doctor_Password;
        public $Doctor_Image;
        public $Doctor_Specialization;

        public function __construct(){
            
        }

        public function validate_params($value){
            if(!empty($value)){
                return true;
            }else{
                return false;
            }
        }

        //Get Doctor List
        public function all_doctors(){
            global $database;
            if($database->escape_value($this->Doctor_Specialization) == "All"){
                $sql = "SELECT * FROM $this->table";
            }else{
                $sql = "SELECT * FROM $this->table WHERE doctor_department = '" . $database->escape_value($this->Doctor_Specialization) . "'";
            }
            $result = $database->query($sql);
            return $database->fetch_array($result);
        }

        public function update_doctor(){
            global $database;

            $this->Doctor_ID = trim(htmlspecialchars(strip_tags($this->Doctor_ID)));
            $this->Doctor_Name = trim(htmlspecialchars(strip_tags($this->Doctor_Name)));
            $this->Doctor_Email = trim(htmlspecialchars(strip_tags($this->Doctor_Email)));
            $this->Doctor_Address = trim(htmlspecialchars(strip_tags($this->Doctor_Address)));
            $this->Doctor_Contact = trim(htmlspecialchars(strip_tags($this->Doctor_Contact)));
            $this->Doctor_Country = trim(htmlspecialchars(strip_tags($this->Doctor_Country)));
            $this->Doctor_State = trim(htmlspecialchars(strip_tags($this->Doctor_State)));
            $this->Doctor_Image = trim(htmlspecialchars(strip_tags($this->Doctor_Image)));


            if($this->Doctor_Image == ""){
                $sql = "UPDATE $this->table SET 
                doctor_name = '" . $database->escape_value($this->Doctor_Name) . "', 
                doctor_address = '" . $database->escape_value($this->Doctor_Address) . "', 
                doctor_contact = '" . $database->escape_value($this->Doctor_Contact) . "',
                doctor_country = '" . $database->escape_value($this->Doctor_Country) . "', 
                doctor_state = '" . $database->escape_value($this->Doctor_State) . "' 
                WHERE doctor_id = '". $database->escape_value($this->Doctor_ID) ."'";
            }else{
                $sql = "UPDATE $this->table SET 
                doctor_name = '" . $database->escape_value($this->Doctor_Name) . "', 
                doctor_address = '" . $database->escape_value($this->Doctor_Address) . "', 
                doctor_contact = '" . $database->escape_value($this->Doctor_Contact) . "',
                doctor_country = '" . $database->escape_value($this->Doctor_Country) . "', 
                doctor_state = '" . $database->escape_value($this->Doctor_State) . "', 
                doctor_image = '" . $database->escape_value($this->Doctor_Image) . "'
                WHERE doctor_id = '". $database->escape_value($this->Doctor_ID) ."'";
            }

            $doctor_saves = $database->query($sql);

            if($doctor_saves){
                return true;
            }else{
                return false;
            }
        }

        public function doctor_profile(){
            global $database;

            $this->Doctor_ID = trim(htmlspecialchars(strip_tags($this->Doctor_ID)));

            $sql = "SELECT * FROM $this->table WHERE doctor_id = '". $database->escape_value($this->Doctor_ID) ."'";

            $result = $database->query($sql);
            $doctor = $database->fetch_row($result);

            if(empty($doctor)){
                return "Doctor doesn't exisit";
            }else{
                return $doctor;
            }
        }

    }

    $doctor = new Doctor();
?>

File: app/lib/Models/doctor.dart
import 'dart:convert';

Doctor doctorModalFromJson(String? str) => Doctor.fromJson(json.decode(str!));

String doctorModalToJson(Doctor data) => json.encode(data.toJson());

class Doctor {
  Doctor({
    this.doctorId,
    this.doctorName,
    this.doctorEmail,
    this.doctorContact,
    this.doctorCountry,
    this.doctorState,
    this.doctorAddress,
    this.doctorSpecialization,
    this.doctorExperience,
    this.doctorImage,
    this.userToken,
  });

  String? doctorId;
  String? doctorName;
  String? doctorEmail;
  String? doctorContact;
  String? doctorCountry;
  String? doctorState;
  String? doctorAddress;
  String? doctorSpecialization;
  String? doctorExperience;
  String? doctorImage;
  String? userToken;

  Doctor.fromMap(Map<dynamic, dynamic> map){
    doctorId = map['doctor_id'];
    doctorName = map["doctor_name"];
    doctorEmail = map["doctor_email"];
    doctorContact = map["doctor_contact"];
    doctorCountry = map["doctor_country"];
    doctorState = map["doctor_state"];
    doctorAddress = map["doctor_address"];
    doctorSpecialization = map["doctor_department"];
    doctorExperience = map["doctor_experience"];
    doctorImage =  map["doctor_image"];
    userToken = map["userToken"];
  }
  factory Doctor.fromJson(Map<dynamic, dynamic> json) => Doctor(
    doctorId: json["doctor_id"],
    doctorName: json["doctor_name"],
    doctorEmail: json["doctor_email"],
    doctorContact: json["doctor_contact"],
    doctorCountry: json["doctor_country"],
    doctorState: json["doctor_state"],
    doctorAddress: json["doctor_address"],
    doctorSpecialization: json["doctor_department"],
    doctorExperience: json["doctor_experience"],
    doctorImage: json["doctor_image"],
    userToken: json["userToken"],
  );

  Map<String, dynamic> toJson() => {
    "doctor_id": doctorId,
    "doctor_name": doctorName,
    "doctor_email": doctorEmail,
    "doctor_contact": doctorContact,
    "doctor_country": doctorCountry,
    "doctor_state": doctorState,
    "doctor_address": doctorAddress,
    "doctor_department": doctorSpecialization,
    "doctor_experience": doctorExperience,
    "doctor_image": doctorImage,
    "userToken": userToken,
  };
}